Doc/library/test.rst

```diff
@@ -676,3 +676,196 @@ The :mod:`test.support` module defines the following classes:
+
+   This will also include stderr output if the process was started with
+   :func:`spawn_python`.
+
+
+.. function:: make_script(script_dir, script_basename, \
+                          source, omit_suffix=False)
+
+   Creates a new Python script/module in the given directory.
+
+   This function invalidates the import system caches, allowing the created
+   file to be immediately imported.
+
+   By default, *script_dir* and *script_basename* are combined with
+   :data:`os.extsep` and the normal ``py`` extension to create the full
+   path to the file. If *omit_suffix* is true, the base name is used
+   directly as the name of created file with no extension.
+
+   The *source* argument is a Unicode string which will be written to the file
+   as UTF-8.
+
+
+.. function:: make_zip_script(zip_dir, zip_basename, \
+                              script_name, name_in_zip=None)
+
+   Creates a new zip archive in the given directory, containing the specified
+   Python script/module.
+
+   This function invalidates the import system caches, allowing the created
+   file to be immediately imported.
+
+   By default, the script is placed at the base of the zip archive using
+   just the same filename as the script itself. This can be changed by
+   specifying the *name_in_zip* parameter to choose a particular name.
+
+   There is also a special case for when *script_name* refers to a file
+   in a ``__pycache__`` directory. In that case, the helper will create a
+   new compiled bytecode file from the original source file using the
+   legacy naming scheme and place that in the zip archive instead.
+
+
+.. function:: make_pkg(pkg_dir, init_source='')
+
+   Creates a simple self-contained Python package, optionally with a
+   non-empty init file.
+
+   This function invalidates the import system caches, allowing the created
+   directory to be immediately imported.
+
+   Equivalent to::
+
+      os.mkdir(pkg_dir)
+      make_script(pkg_dir, "__init__", init_source)
+
+
+.. function:: make_zip_pkg(zip_dir, zip_basename, \
+                           pkg_name, script_basename, source, \
+                           depth=1, compiled=False)
+
+   Create a self-contained Python package inside a new zip archive in the
+   given directory.
+
+   This function invalidates the import system caches, allowing the contents
+   of the created archive to be immediately imported.
+
+   For example, this call::
+
+      make_zip_pkg('.', 'example', 'mypkg', 'submodule', '', depth=2)
+
+   Would create an archive in the current directory called ``example.zip``
+   with the layout::
+
+      mypkg/
+        __init__.py
+        mypkg/
+          __init__.py
+          submodule.py
+
+   The init modules in the packages inside the archive are always empty,
+   while *source* is written to the submodule named by *script_basename*.
+
+   The *depth* argument controls how many deeply nested the package is.
```

Comment:
. . . controls how deeply nested the package is.

or

. . . controls how many packages the submodule is nested inside.